src/nccl_ofi_rdma.c

```diff
@@ -223,32 +223,50 @@ static inline nccl_net_ofi_rdma_recv_comm_t *get_recv_comm(nccl_net_ofi_rdma_dev
  * Get connection message from bounce buffer
  */
 static inline nccl_ofi_rdma_connection_info_t *get_bounce_connection_msg(
-	nccl_net_ofi_rdma_bounce_fl_item_t *bounce_fl_item)
+	rdma_req_bounce_data_t *bounce_data)
 {
-	return (nccl_ofi_rdma_connection_info_t *)&bounce_fl_item->bounce_msg;
+	return (nccl_ofi_rdma_connection_info_t *)bounce_data->bounce_fl_elem->ptr;
 }
 
 /*
  * Get ctrl message from bounce buffer
  */
 static inline nccl_net_ofi_rdma_ctrl_msg_t *get_bounce_ctrl_msg
-	(nccl_net_ofi_rdma_bounce_fl_item_t *bounce_fl_item)
+	(rdma_req_bounce_data_t *bounce_data)
 {
-	return (nccl_net_ofi_rdma_ctrl_msg_t *)&bounce_fl_item->bounce_msg;
+	return (nccl_net_ofi_rdma_ctrl_msg_t *)bounce_data->bounce_fl_elem->ptr;
 }
 
 /*
  * Get close message from bounce buffer
  */
 static inline nccl_net_ofi_rdma_close_msg_t *bounce_get_close_msg
-	(nccl_net_ofi_rdma_bounce_fl_item_t *bounce_fl_item)
+	(rdma_req_bounce_data_t *bounce_data)
 {
 	nccl_net_ofi_rdma_close_msg_t *close_msg =
-		(nccl_net_ofi_rdma_close_msg_t *)&bounce_fl_item->bounce_msg;
+		(nccl_net_ofi_rdma_close_msg_t *)bounce_data->bounce_fl_elem->ptr;
 	assert(close_msg->type == NCCL_OFI_RDMA_MSG_CLOSE);
 	return close_msg;
 }
 
+/**
+ * Get ctrl message from send_ctrl_data
+ */
+static inline nccl_net_ofi_rdma_ctrl_msg_t *send_ctrl_get_msg
```

Comment:
these should not be inline.  However, they should be prefixed with `rdma`, so that they don't violate the C++ uniqueness rule with the sendrecv protocol.

Reply:
> these should not be inline.

Why? They are small enough functions that I wouldn't want them to be not-inlined.

> However, they should be prefixed with `rdma`

👍

Reply:
The compiler is going to inline them anyway for these non-side effect accessor functiokns.